### PR TITLE
No longer rely on iife for scoping inside loops

### LINKED ISSUE #1540: [FEATURE REQUEST] rewrite closure compilation inside for loops
Use `let/const` instead of iife

### PR DESCRIPTION
**Pull request by hhugo**
Fix #1540

### CODE REVIEW

File: compiler/tests-compiler/gh1007.ml [hunk 1]
```diff
@@ -158,166 +158,162 @@ let ()  = M.myfun M.x
...
-                   : 0
-                     < caml_int_compare(x2$0, x3)
-                     ? [0, x2$0, [0, x3, [0, x1$0, 0]]]
-                     : [0, x3, [0, x2$0, [0, x1$0, 0]]];
-             return [0, s$0, tl$1];
-            }
-           }
-          }
+      var next = x$0[1];
+      function sort(n, l){
```
Comment (vouillon) [inline compiler/tests-compiler/gh1007.ml:162]: It seems that the function order got reversed, which makes thinks harder to compare.

File: compiler/tests-compiler/mutable_closure.ml [hunk 2]
```diff
@@ -120,47 +120,39 @@ let%expect_test _ =
...
+        var counter$1 = counter + 1 | 0;
+        return f$0(counter$1, _h_);
+       }
+       if(0 === n) return i$0;
+       var _i_ = n - 1 | 0;
+       if(counter >= 50) return caml_trampoline_return(f$0, [0, _i_]);
+       var counter$0 = counter + 1 | 0;
+       return f$0(counter$0, _i_);
+      }
+      function f(n){return caml_trampoline(f$0(0, n));}
```
[the remark this answers is not in the document]
Comment (hhugo) [reply compiler/tests-compiler/mutable_closure.ml:150]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/block#block_scoping_rules_with_let_const_class_or_function_declaration_in_strict_mode

File: compiler/tests-compiler/gh1007.ml (hunk 1 above)
Comment (hhugo) [reply compiler/tests-compiler/gh1007.ml:162]: The ordering is altered by compiler/lib/strongly_connected_components.ml.
It should fix it both on master and in this PR.

Comment (hhugo) [reply compiler/tests-compiler/gh1007.ml:162]: now fixed

**Pull request merged**